bigtable/integration_test.go

```diff
@@ -899,6 +915,51 @@ func TestIntegration_LargeReadsWritesAndScans(t *testing.T) {
 	} else if status[0] == nil || status[1] == nil {
 		t.Fatalf("No error for bad bulk mutation")
 	}
+
+	// Validate that metrics are exported
+	elapsedTime := time.Since(testStartTime)
+	if elapsedTime < 2*defaultSamplePeriod {
+		// Ensure at least 2 datapoints are recorded
+		time.Sleep(2*defaultSamplePeriod - elapsedTime)
+	}
+
+	timeListEnd := time.Now()
+	tsListEnd := &timestamppb.Timestamp{
+		Seconds: timeListEnd.Unix(),
+		Nanos:   int32(timeListEnd.Nanosecond()),
+	}
+
+	monitoringClient, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		t.Errorf("Failed to create metric client: %v", err)
+	}
+
+	metricNamesValidate := []string{
+		metricNameOperationLatencies,
+		metricNameAttemptLatencies,
+		metricNameServerLatencies,
+	}
+	for _, metricName := range metricNamesValidate {
+		// ListTimeSeries can list only one metric type at a time.
+		// So, call ListTimeSeries with different metric names
+		iter := monitoringClient.ListTimeSeries(ctx, &monitoringpb.ListTimeSeriesRequest{
```

Comment:
I think this will be flaky because because the data might not be immediately available. I would do an initial sleep for 30s and then poll for data every 10s for 5 mins

Also, can you extract this to its own integration test .. this one seems to be getting out of hand

Reply:
Done